lua/typescript-tools/protocol/text_document/completion/resolve.lua

```diff
@@ -137,6 +137,12 @@ end
 function M.handler(request, response, params)
   local requested_bufnr = vim.uri_to_bufnr(vim.uri_from_fname(params.data.file))
   local filetype = vim.bo[requested_bufnr].filetype
+  local is_valid_context_for_function_snippet = utils.is_valid_context_for_function_snippet(
+    requested_bufnr,
+    params.data,
```

Comment:
After using this PR for a bit, I've noticed that the position here does under some circumstances not get updated between when it first appears as a completion item and when it is chosen as a completion.
For example, when you are in insert mode and press backspace, the results and positions remain the same, which can lead to some unexpected behavior like this (| indicates cursor position):

```
# go into insert mode
{fooBarBaz|}<Component></Component>
# press backspace once, if you complete now, a snippet is used correctly
{fooBarBa|}<Component></Component>
# press backspace two more times, if you complete now, a snippet is not used
# even though it should be
# because the position is still the same as in step 1, but there is a '<'
# on that position now
{fooBar|}<Component></Component>
```

The problem could be solved, by using the cursor position from nvim, instead of the params here, but I'm not sure if there is a better solution to this.